### DISCUSSION TITLE
Using a non-public Azure model

### DISCUSSION THREAD

**Discussion by stefanlundchristensen** [logancyang/obsidian-copilot#203] [Q&A]
Hi team,

I have access to a private Azure instance where there is a nice model. Would it be possible to drop this into this plug-in and if so - how?

Cheers,
Stefan

**Comment by logancyang** [comment]
Hi @stefanlundchristensen, you can use your Azure credentials in Copilot settings, make sure your credentials are correct and test it using `curl` first

<img width="808" alt="Screenshot 2023-12-12 at 4 32 47 PM" src="https://github.com/logancyang/obsidian-copilot/assets/4860545/b9738c6b-0513-49d5-ac91-22ba59317bcd">

Test with something like (the exact format might be different)
```
curl https://<your-instance>.openai.azure.com/openai/deployments/<your-deployment>/completions\?api-version\=<your-api-version>\
  -H "Content-Type: application/json" \
  -H "api-key: <your-api-key>" \
  -d "{
  \"prompt\": \"Once upon a time\",
  \"max_tokens\": 5
}"
```

I did not get the parameters right on the first try and asked for Azure support, they are quite responsive and helpful.

**Comment by stefanlundchristensen** [comment]
Perfect thanks! I have something that is working now. Do get an error like below but seems to be working anyway

`LangChain error: TypeError: Cannot read properties of undefined (reading 'text') `

**Reply by logancyang** [reply]
@stefanlundchristensen try putting some random string in OpenAI API key? This should be a known issue and will be fixed soon, sorry about that
